### ISSUE TITLE
fastcorrect 效果没有复现

### ISSUE THREAD

**Issue by nieallen** [microsoft/NeuralSpeech#14]
您好，我没有复现论文效果呢，加载您仓库里提供的预训练模型，用aishell数据做的微调。loss最终在2.34左右，但to_be_edited_loss下降不太正常，一直都是0.33-0.32

**Comment by nieallen** [comment]
>

**Comment by YichongLeng** [comment]
How many epoch the model is finetuned? In my experiment, after finetuning 15 epoch I achieve the WERR in paper.

**Comment by YichongLeng** [comment]
I did not save the finetuning log. I will re-run the finetuning and maybe I can share my log to you for your reference.

**Comment by nieallen** [comment]
> I did not save the finetuning log. I will re-run the finetuning and maybe I can share my log to you for your reference.

Thanks! Could you maybe explain what to_be_edited_loss means?

**Comment by YichongLeng** [comment]
> > I did not save the finetuning log. I will re-run the finetuning and maybe I can share my log to you for your reference.
> 
> Thanks! Could you maybe explain what to_be_edited_loss means?

During the training stage, we add a binary CE loss on each source token to let the model learn to predict whether a token is correct or not. This CE loss will help the learning of duration predictor. Please note that the trained binary classifier will not be used during the inference so this loss is an auxiliary loss.

**Comment by YichongLeng** [comment]
@nieallen I quickly finetune the released pretrain model for 20 epoch and test the finetuned model, whose performance is normal. You can check my log in https://drive.google.com/file/d/1X2ydsXU99c-ZwZWUzTKixeI1L35NNINo/view?usp=sharing.

BTW, there are some ideas from me, which might help you:
1. I finetune the pretrained model on 4 GPUs instead of 1 GPU.
2. Since AISHELL-1 is a small dataset, you cannot finetune it for too much epoch (maybe 15-20 epoch is enough).

Feel free to let me know if you still meet problem.

**Comment by Amber921463001** [comment]
> @nieallen I quickly finetune the released pretrain model for 20 epoch and test the finetuned model, whose performance is normal. You can check my log in https://drive.google.com/file/d/1X2ydsXU99c-ZwZWUzTKixeI1L35NNINo/view?usp=sharing.
> 
> BTW, there are some ideas from me, which might help you:
> 
> 1. I finetune the pretrained model on 4 GPUs instead of 1 GPU.
> 2. Since AISHELL-1 is a small dataset, you cannot finetune it for too much epoch (maybe 15-20 epoch is enough).
> 
> Feel free to let me know if you still meet problem.

请问您用4块gpu训练是设置update-freq=4吗？因为我看网上说在单个GPU上指定“--update-freq 4”来训练，结果和4个GPU训练是基本等价的

**Comment by YichongLeng** [comment]
I not only use 4 GPU, but also set update-freq=4. So if you just finetune FastCorrect on one card, you need to set update-freq=16.
From my log you can know all the hyper-parameters, you can have a check with yours.

**Comment by dahu1** [comment]
> How many epoch the model is finetuned? In my experiment, after finetuning 15 epoch I achieve the WERR in paper.

How many epoch the model is pretrained ?
